cpp/llm_chat.cc

```diff
@@ -394,25 +395,44 @@ class LLMChat {
   }
 
   /**
-   * Get input tokens based on history
+   * Get prompt based on history
+   */
+  std::vector<std::string> GetPrompt() {
+    if (this->total_seq_len_ == 0) {
+      return this->conversation_.GetPromptArray();
+    }
+    return this->conversation_.GetPromptArrayLastRound();
+  }
+
+  /**
+   * Get input tokens based on history. Can pass in string prompt optionally.
    */
-  std::vector<int32_t> GetInputTokens() {
+  std::vector<int32_t> GetInputTokens(std::optional<std::string> prompt_str = std::nullopt,
+                                      bool add_bos_token = false) {
     std::vector<int32_t> tokens;
     std::vector<std::string> prompts;
-
-    if (this->total_seq_len_ == 0) {
-      prompts = this->conversation_.GetPromptArray();
-      if (this->conversation_.add_bos) {
-        tokens.insert(tokens.begin(), bos_token_id_);
-      }
+    std::string all_prompt;
+    if (prompt_str.has_value()) {
+      all_prompt = prompt_str.value();
     } else {
-      prompts = this->conversation_.GetPromptArrayLastRound();
+      prompts = GetPrompt();
+      all_prompt = GetConcatPrompt(prompts, 0, 0);
+    }
+    std::cout << "all prompt: " << all_prompt << std::endl;
```

Comment:
Is these `std::cout` used for debugging? If so we should either delete them or use `LOG(DEBUG) << ` instead of the standard output stream.

Reply:
removed